### DISCUSSION TITLE
dynamic topic kafka

### DISCUSSION THREAD

**Discussion by taquanghung1705199** [vectordotdev/vector#14429] [Q&A] [answered]
I have a question that:
I'm source from http and sink to kafka

I want when I send log to http have path is: /<name> => log is sent to kafka with topic <name>
Ex:
source /aaa => topic aaa
source /bbb => topic bbb
source /ccc => topic ccc

How to do this ?

**Comment by spencergilbert** [comment, marked as answer]
👋 
Hi @taquanghung1705199, the [`topic`](https://vector.dev/docs/reference/configuration/sinks/kafka/#topic) is template-able, you would set it with a template syntax that resolves to the path.

Ex, with the default [`path_key`](https://vector.dev/docs/reference/configuration/sources/http/#path_key) of "path":

```toml
topic = "{{ path }}"
```

**Reply by taquanghung1705199** [reply]
thanks
I got it!
